### PR TITLE
Added legacy cluster types to the assessment result

### LINKED ISSUE #925: [BUG]: UCX assessment does not list No-Isolation Shared Clusters
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

No-Isolation shared clusters are not compatible with Unity Catalog. During the assessment, No isolation shared clusters should be listed in the assessment summary dashboard which is not happening at the moment

### Expected Behavior

No isolation shared clusters should be listed in the assessment summary dashboard

### Steps To Reproduce

Create a non-isolation shared cluster and run the assessment

### Cloud

Azure

### Operating System

macOS

### Version

latest via Databricks CLI

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by william-conti**
## Changes
Adding missing support in the assessment for no isolation shared and legacy cluster types

### Linked issues
Resolves #925

### Tests

- [X] manually tested
- [x] added unit tests
- [X] added integration tests
- [ ] verified on staging environment (screenshot attached)

### CODE REVIEW

**Review by nfx — APPROVED** [review]
Lgtm, but make sure that pr title is changelog-ready (eg very explicit) and you "resolves ..." the correct issue

File: tests/integration/assessment/test_clusters.py [hunk 1]
```diff
@@ -22,3 +23,18 @@ def test_cluster_crawler(ws, make_cluster, inventory_schema, sql_backend):
 
     assert len(results) >= 1
     assert results[0].cluster_id == created_cluster.cluster_id
+
+
+def test_cluster_crawler_no_isolation(ws, make_cluster, inventory_schema, sql_backend):
+    created_cluster_2 = make_cluster(data_security_mode=DataSecurityMode.NONE, num_workers=1)
```
Comment (nkvuong) [inline tests/integration/assessment/test_clusters.py:29]: do we need the suffix `_2`

```suggestion
    created_cluster = make_cluster(data_security_mode=DataSecurityMode.NONE, num_workers=1)
```

File: tests/integration/assessment/test_clusters.py [hunk 2]
```diff
@@ -22,3 +23,18 @@ def test_cluster_crawler(ws, make_cluster, inventory_schema, sql_backend):
...
+    cluster_crawler = ClustersCrawler(ws=ws, sbe=sql_backend, schema=inventory_schema)
+    clusters = cluster_crawler.snapshot()
+    results = []
+    for cluster in clusters:
+        if cluster.success != 0:
+            continue
+        if cluster.cluster_id == created_cluster_2.cluster_id:
+            results.append(cluster)
+
+    assert len(results) >= 1
```
Comment (nkvuong) [inline tests/integration/assessment/test_clusters.py:39]: shouldn't there be exactly 1 result?

```suggestion
    assert len(results) = 1
```

File: tests/integration/assessment/test_clusters.py [hunk 3]
```diff
@@ -22,3 +23,18 @@ def test_cluster_crawler(ws, make_cluster, inventory_schema, sql_backend):
...
+
+
+def test_cluster_crawler_no_isolation(ws, make_cluster, inventory_schema, sql_backend):
+    created_cluster_2 = make_cluster(data_security_mode=DataSecurityMode.NONE, num_workers=1)
+    cluster_crawler = ClustersCrawler(ws=ws, sbe=sql_backend, schema=inventory_schema)
+    clusters = cluster_crawler.snapshot()
+    results = []
+    for cluster in clusters:
+        if cluster.success != 0:
+            continue
```
Comment (nkvuong) [inline tests/integration/assessment/test_clusters.py:35]: do we need this check?

File: tests/unit/assessment/test_clusters.py [hunk 4]
```diff
@@ -265,3 +266,37 @@ def test_try_fetch():
...
+
+def test_no_isolation_clusters():
+    sample_clusters = [
+        ClusterDetails(
+            cluster_name="No isolation shared",
+            spark_version="12.3.x-cpu-ml-scala2.12",
+            data_security_mode=DataSecurityMode.NONE,
+        )
+    ]
+    ws = workspace_client_mock()
```
Comment (nkvuong) [inline tests/unit/assessment/test_clusters.py:279]: can we load the cluster config from a json file, similar to other cluster unit test?

File: src/databricks/labs/ucx/assessment/clusters.py [hunk 5]
```diff
@@ -111,6 +117,10 @@ def check_cluster_failures(self, cluster: ClusterDetails, source: str) -> list[s
             failures.extend(self._check_cluster_policy(cluster.policy_id, source))
         if cluster.init_scripts is not None:
             failures.extend(self._check_cluster_init_script(cluster.init_scripts, source))
+        if cluster.data_security_mode == DataSecurityMode.NONE:
+            failures.append("No isolation shared clusters not supported in UC")
```
Comment (nkvuong) [inline src/databricks/labs/ucx/assessment/clusters.py:121]: can we add this error message to [doc](https://github.com/databrickslabs/ucx/docs/assessment.md) as well

**Review by nkvuong — CHANGES_REQUESTED** [review]
looks fine to me, bar a couple of suggested changes

File: tests/unit/assessment/test_clusters.py (hunk 4 above)
Comment (william-conti) [reply tests/unit/assessment/test_clusters.py:279]: The test is relatively simple and doesn't really need to have a JSON file similar to the other tests that carries more complex configurations and cases

**Pull request merged**